app/models/miq_worker/systemd_common.rb

```diff
@@ -0,0 +1,144 @@
+require "dbus/systemd"
+
+class MiqWorker
+  module SystemdCommon
+    extend ActiveSupport::Concern
+
+    class_methods do
+      def ensure_systemd_files
+        File.write(target_file_path, target_file) unless target_file_path.exist?
+        File.write(service_file_path, unit_file) unless service_file_path.exist?
+        Dir.mkdir(service_config_path) unless service_config_path.exist?
```

Comment:
You might want `FileUtils.mkdir_p` instead

Reply:
Minor, but if all of the path helper methods return Pathname objects, then you can chain the writes as well.  e.g. `target_file_path.write(target_file) unless target_file_path.exist?`